diff --git a/aosp_redfin.mk b/aosp_redfin.mk
index 25756bc..7ede74b 100644
--- a/aosp_redfin.mk
+++ b/aosp_redfin.mk
@@ -43,8 +43,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/handheld_vendor.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/telephony_vendor.mk)
 
 $(call inherit-product, device/google/redfin/device-redfin.mk)
-$(call inherit-product-if-exists, vendor/google_devices/redfin/proprietary/device-vendor.mk)
-$(call inherit-product-if-exists, vendor/google_devices/redfin/prebuilts/device-vendor-redfin.mk)
+$(call inherit-product-if-exists, vendor/google/redfin/proprietary/device-vendor.mk)
 
 # Exclude features that are not available on AOSP devices.
 PRODUCT_COPY_FILES += \
